src/rust/grapl-web-ui/src/authn/user.rs

```diff
@@ -25,18 +24,16 @@ pub struct AuthenticatedUser {
 
 #[derive(thiserror::Error, Debug)]
 pub enum UserAuthenticationError {
-    #[error(transparent)]
-    Client(#[from] AuthDynamoClientError),
-    #[error(transparent)]
-    Cookie(#[from] actix_web::Error),
-    #[error("Session token not found in user cookie")]
-    NoSessionToken,
-    #[error("Unable to get database client from Actix app data")]
-    AddData,
-    // This Opaque variant is a hack around mixing tokio runtimes and
-    // ResponseError not implementing Send
-    #[error("Opaque error from DB client")]
-    Opaque,
+    #[error("unable to validate user session")]
+    SessionValidation(#[source] AuthDynamoClientError),
+    #[error("unable to get user data from validated session")]
+    UserNotFound(#[source] AuthDynamoClientError),
+    #[error("unable to access session storage")]
+    SessionStorage(#[source] actix_web::Error),
+    #[error("user is not authenticated")]
+    Unauthenticated,
+    #[error("unable to get database client from Actix app data")]
+    DynamoClientUnavailable,
```

Comment:
Why are all these errors different if we're just updating Actix?

Reply:
> Why are all these errors different if we're just updating Actix?

Primarily due to dropping `UserAuthenticationError::Opaque`, which was part of the ugly hack of mixing tokio runtimes (see variant comment). While changing that variant and the function using this error type, these other variants stood out as desperately needing improvement as well.

I plan on making similar improvements to all other error handling in the crate, but I'm saving that for another PR.